### ISSUE TITLE
EMail Server Settings SMTP Port 587 SSL Not connecting

### ISSUE THREAD

**Issue by tmcphail67** [btcpayserver/btcpayserver#3849]
Server and Store Email settings.  I am receiving the error "A task was canceled" when testing my SMTP mail server settings using port 587 with ssl.  When I set up port 25 without ssl the test email sends no porblem.  I am seeing the following in the hmailserver logs:

"TCPIP"	9740	"2022-06-09 13:04:24.353"	"TCP - 192.168.1.1 connected to 192.168.1.11:587."
"TCPIP"	9740	"2022-06-09 13:04:33.979"	"TCPConnection - TLS/SSL handshake failed. Session Id: 570, Remote IP: 192.168.1.1, Error code: 1, Message: stream truncated"

![Capture](https://user-images.githubusercontent.com/61200660/172916610-c2046866-ba56-4f35-b8d4-4bfa2e1de5a0.PNG)

- BTCPay Server Version:    1.5.4.0
- Deployment Method:        Ubuntu Docker
- Browser:                  Firefox

Thank you!

**Comment by tmcphail67** [comment]
Thanks for your response!

I have not figured it out.  I have tried two different SMTP servers with the same result.  It probably has something to do with the SSL/TLS configuration.  One of the servers I used was set up for STARTTLS and the other I was able to set it to SSL/TLS or STARTTLS with the same result.  With the second server it is connecting fine without encryption.

The thunderbird mail client is connecting on port 587 with the following connection parameters.

![Screenshot 2022-06-14 150128](https://user-images.githubusercontent.com/61200660/173678439-b61fa0d0-c235-4809-987f-33f6567c0c49.jpg)

So without any guidance from you on what the BTCPay Server is looking for I don't have a clue what the issue is.

Thanks for your help!
